app/models/notification.rb

```diff
@@ -371,13 +371,15 @@ def self.populate_acting_user(notifications)
           (
             notification.data_hash[:username] || notification.data_hash[:display_username] ||
               notification.data_hash[:mentioned_by_username] ||
-              notification.data_hash[:invited_by_username]
+              notification.data_hash[:invited_by_username] ||
+              notification.data_hash[:original_username]
           )&.downcase
       end
 
     users = User.where(username_lower: usernames.uniq).index_by(&:username_lower)
     notifications.each do |notification|
       notification.acting_user = users[notification.acting_username]
+      notification.data_hash[:original_name] = notification.acting_user&.name
```

Comment:
Don't we have scenarios where we don't want to expose the users names? I _think_ `enable names` `false` is one of them, but there may be others...

Reply:
oh interesting. these are situations where the profile page allows name creation, but it needs to be always hidden in the UI?

Reply:
I added a check/test for enable_names. Also looked through where enable_names is used in the codebase. Its used extensively and doesn't seem to be paired with any other site setting that would be relevant here. @renato